akka-stream/src/main/scala/akka/stream/scaladsl/TLS.scala

```diff
@@ -76,11 +76,14 @@ object TLS {
       sslConfig.getOrElse(AkkaSSLConfig(system))
 
     val createSSLEngine = { system: ActorSystem ⇒
+      val config = theSslConfig(system)
+
       val engine = hostInfo match {
-        case Some((hostname, port)) ⇒ sslContext.createSSLEngine(hostname, port)
-        case None                   ⇒ sslContext.createSSLEngine()
+        case Some((hostname, port)) if !config.config.loose.disableSNI ⇒
+          sslContext.createSSLEngine(hostname, port)
```

Comment:
You mentioned you had a test case that demonstrated SNI got disabled but hostname verification was still functional - could you turn that into a unit/integrationtest?

Reply:
I could provide the test, the problem is it relies on a weirdly configured local server that I won't be able to keep publicly accessible.

And I don't think I could make a standalone test case that simulates the SNI issues without this =/